sound/soc/intel/boards/sof_sdw.c

```diff
@@ -515,6 +515,44 @@ int sdw_trigger(struct snd_pcm_substream *substream, int cmd)
 	return ret;
 }
 
+int sdw_hw_params(struct snd_pcm_substream *substream,
+		  struct snd_pcm_hw_params *params)
+{
+	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
+	struct snd_soc_dai *cpu_dai;
+	struct snd_soc_dai *codec_dai;
+	unsigned int ch_mask;
+	int step;
+	int i;
+	int j;
+
+	if (!rtd->dai_link->codec_ch_maps)
+		return 0;
+
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
+		step = 0;
+		ch_mask = GENMASK(params_channels(params) - 1, 0);
+	} else {
+		step = rtd->dai_link->num_codecs;
+		ch_mask = GENMASK(params_channels(params) / step - 1, 0);
```

Comment:
why we have this distinction?
The playback could in theory have more than one codec, in case of 1 to 1 connection we will have the same result, no?

Reply:
> why we have this distinction? The playback could in theory have more than one codec, in case of 1 to 1 connection we will have the same result, no?

Currently, we send identical data to all codecs in playback. For example, codec A is on sdw link 0 and codec B is on sdw link 1, we will send identical data to codec A and B, So, we want to have the same ch_mask for all codecs.
We can change the ch_mask to split the cpu channels in the future.

Reply:
Right, OK. I would add a comment to make sure that it is recorded.

The other thing that I'm wondering about is: What happens if you start a capture in stereo? you will record one  channel from codec A and one from codec B?

Reply:
> Right, OK. I would add a comment to make sure that it is recorded.

Ok, I will add a comment.

> 
> The other thing that I'm wondering about is: What happens if you start a capture in stereo? you will record one channel from codec A and one from codec B?

Yes.